File: others/프로그래머스_카펫.py
def solution(brown, yellow):
    answer = []

    total = brown + yellow

    for i in range(1, total // 2):
        # 나누어떨어진다면,
        if total % i == 0:
            column = i  # 세로
            row = total // i  # 가로

            if (column-2) * (row-2) == yellow:
                answer = [row, column]
                break

    return answer

File: others/프로그래머스_카펫.js
function solution(brown, yellow) {
    let answer = [];

    brown = parseInt(brown)
    yellow = parseInt(yellow)

    const total = brown + yellow

    for (let i = 1; i < parseInt(total / 2); i++) {
        if (total % i == 0) {
            const column = i
            const row = parseInt(total / i)

            if ( (column-2) * (row-2) === yellow) {
                answer = [column, row]
                break
            }
        }
    }

    return answer;
}